app/assets/javascripts/search.js

```diff
@@ -17,4 +18,42 @@ pages.search.hideForm = function(){
       }
     });
   }
-}
+};
+
+pages.search.toggleOptions = function(){
+  $('.js-toggle-archive-options').click(function(){
+    $('.js-archive-options').toggle();
+    $('.js-toggle-archive-options').toggleClass('active')
```

Comment:
Just a minor thing maybe, but I would call the `active` class as `is-active` and put it in a state.scss file. And there, you can have all your css classes that are added by javascript. Like `is-hidden`, `is-active` an so.